src/pykeen/triples/triples_factory.py

```diff
@@ -359,11 +359,14 @@ def _add_inverse_triples_if_necessary(self, mapped_triples: MappedTriples) -> Ma
 
     def create_slcwa_instances(self) -> Instances:
         """Create sLCWA instances for this factory's triples."""
-        return SLCWAInstances(mapped_triples=self._add_inverse_triples_if_necessary(mapped_triples=self.mapped_triples))
+        return self._create_instances(SLCWAInstances)
 
     def create_lcwa_instances(self, use_tqdm: Optional[bool] = None) -> Instances:
         """Create LCWA instances for this factory's triples."""
-        return LCWAInstances.from_triples(
+        return self._create_instances(LCWAInstances)
+
+    def _create_instances(self, instances_cls: Type[Instances]) -> Instances:
+        return instances_cls.from_triples(
```

Comment:
If we made this the constructor instead of a class method, we could use a class resolver.

Reply:
Yes, but I don't want to make this PR more complicated